### PR TITLE
Describe read-only mode of ingesters.

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR describes read-only mode of ingesters, and adds API documentation for `/ingester/prepare-partition-downscale` and `/ingester/prepare-instance-ring-downscale` endpoints.

#### Checklist

- [na] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [na] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Comment by pstibrany** [comment]
I'll wait for @tacole02 review before merging.

File: docs/sources/mimir/references/architecture/components/ingester.md
Comment (jhalterman) [inline docs/sources/mimir/references/architecture/components/ingester.md:103]: Do we want to link to the API doc here?

Comment (pstibrany) [reply docs/sources/mimir/references/architecture/components/ingester.md]: Added

File: docs/sources/mimir/references/http-api/index.md
Comment (tacole02) [inline docs/sources/mimir/references/http-api/index.md:425]:
```suggestion
### Prepare partition downscale
```

File: docs/sources/mimir/references/http-api/index.md
Comment (tacole02) [inline docs/sources/mimir/references/http-api/index.md:431]:
```suggestion
This endpoint prepares the ingester's partition for downscaling by setting it to the `INACTIVE` state.
```

File: docs/sources/mimir/references/http-api/index.md
Comment (tacole02) [inline docs/sources/mimir/references/http-api/index.md:433]:
```suggestion
A `GET` call to this endpoint returns a timestamp of when the partition was switched to the `INACTIVE` state, or 0, if the partition is not in the`INACTIVE` state.
```

**Review by tacole02 — APPROVED** [review]
Looks good! Thank you for updating this doc!

**Comment by pstibrany** [comment]
> Looks good! Thank you for updating this doc!

Thank you for your review and all suggestions!

**Pull request merged**
